source/common/http/conn_pool_grid.cc

```diff
@@ -198,12 +198,13 @@ ConnectivityGrid::ConnectivityGrid(
     const Network::TransportSocketOptionsConstSharedPtr& transport_socket_options,
     Upstream::ClusterConnectivityState& state, TimeSource& time_source,
     AlternateProtocolsCacheSharedPtr alternate_protocols, ConnectivityOptions connectivity_options,
-    Quic::QuicStatNames& quic_stat_names, Stats::Scope& scope)
+    Quic::QuicStatNames& quic_stat_names, Stats::Scope& scope, Http::PersistentQuicInfo& quic_info)
```

Comment:
So now you've done all the plumbing I go and ask the inconvenient questions, but given we agree this is persistent info for the cluster, should we expose it via the cluster API rather than pass it in separately?

host_->cluster()->quicInfo or some such?

Reply:
Does this mean the quic info should live in ClusterInfoImpl object? I might be wrong but it seems to be shared across the workers. QUIC objects like alarm factory are not thread safe.

Reply:
I think we'd get thread local cluster, but given we'd still have to cast from the interface I can go either way here.